api/envoy/api/v2/cds.proto

```diff
@@ -234,6 +234,32 @@ message Cluster {
   //   verification.
   auth.UpstreamTlsContext tls_context = 11;
 
+  // UpstreamTransportSocket specifies the transport socket config used for upstream connection.
+  message UpstreamTransportSocket {
```

Comment:
Maybe this should be used for default `transport_socket` `tls_context` in the cluster as well? Looks clearer to me to use the same proto msg, but that requires xDS management server changes to migrate.

Not sure if it's worth it.

Reply:
That's why I filed #8098, then you shouldn't need the oneof below.